src/libraries/System.Private.CoreLib/src/System/Threading/RegisteredWaitHandle.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+#if !FEATURE_WASM_THREADS
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
+#endif
+    public sealed partial class RegisteredWaitHandle : MarshalByRefObject
+    {
+        private readonly object? _lock;
+        private SafeWaitHandle? _waitHandle;
+        private readonly _ThreadPoolWaitOrTimerCallback? _callbackHelper;
+        private readonly uint _millisecondsTimeout;
+        private bool _repeating;
```

Comment:
The fields of portable and windows thread pool should be de-duplicated.

For example, `_repeating` is duplicated with `bool Repeating { get; }` property..

Reply:
Should I keep the field or the property instead?

Reply:
In complicated cases like this one, it is better to have explicit fields for everything - it makes it easier to see what's going on.